string id
uint32 length
Keyline[] Keylines